compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtInterfaces.td

```diff
@@ -40,4 +41,98 @@ def LinalgExtInterface : OpInterface<"LinalgExtOp"> {
   let verify = [{ return detail::verifyLinalgExtOpInterface($_op); }];
 }
 
+// Interface that allows for fusion of both LinalgExt and Linalg ops.
+def LinalgFusionInterface : OpInterface<"LinalgFusionOpInterface", [DestinationStyleOpInterface]> {
```

Comment:
I think you can drop the `DestinationStyleOpInterface`. I see why you are using it, but it is kind of unnecessary.

Reply:
Okay, then should uses of the `DestinationStyleOpInterface`'s methods just cast from the fusion interface then?

For example, change
https://github.com/IanWood1/iree/blob/0d700fe7d9e011f4e0e085ed970037bc0268cae1/compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp#L655
to
```cpp
for (OpOperand *inputOperand : cast<LinalgFusionOpInterface>(consumerLinalgOp.getDpsInputOperands())) { 
```

Reply:
You can do 
```
auto dstOp = dyn_cast<DestinationStyleOpInterface>(consumerLinalgOp.getOperation());
if (dstOp) {
  for (OpOperand *inputOperand : op.getDpsInputOperands())
}
```

You can cast from one operation to an interface op type if the op implements the interface. So if there is no link between interfaces, then you need to get the base operation and then cast that.